### PR TITLE
feat(address) updating country and state to be pickers with configurable options

### PR DESCRIPTION
**Pull request by amrutha-rajiv**
## **Description**
country and state fields are pickers where options can be configured like so ->
```
state: {
          label: 'State',
          pickerConfig: {
            field: 'value',
            format: '$label',
            options: (countryID, query) => {
              return new Promise((resolve, reject) => {
                resolve(this.getStateOptions(countryID, query));
              });
            },
            getLabels: (value: number) => {
              return new Promise((resolve, reject) => {
                resolve(this.getStateLabel(value));
              });
            }
          }
        },
        countryID: {
          label: 'Country',
          pickerConfig: {
            field: 'value',
            format: '$label',
            options: (query) => {
              return new Promise((resolve, reject) => {
                resolve(this.getCountryOptions(query));
              });
            },
            getLabels: (value: number) => {
              return new Promise((resolve, reject) => {
                resolve(findByCountryId(value));
              });
            }
          }
        },
```
#### **Verify that...**

- [ ] Any related demos where added and `npm start` still works
- [ ] New demos work in `Safari`, `Chrome` and `Firefox`
- [ ] `npm run lint` passes
- [ ] `npm test` passes and code coverage is increased
- [ ] `npm run compile` still works

##### **Screenshots**

### CODE REVIEW

**Comment by jgodi** [comment]
|File (3)|Change (:white_check_mark: +0.05%)|
|---|---|
|FormValidators.ts|:bangbang: -1.75%|
|Address.ts|:white_check_mark: +0.99%|
|FormUtils.ts|:bangbang: -0.08%|

*Please note, individual file changes won't add up to the overall coverage due to number of lines!*

**All new files must be at 80% coverage and old files must NOT decrease in coverage!**

File: src/app/pages/elements/form/FormDemo.ts [hunk 1]
```diff
@@ -338,22 +349,50 @@ export class FormDemoComponent {
...
+          }
         },
-        country: {
+        countryID: {
           label: 'Country',
+          pickerConfig: {
+            field: 'value',
+            format: '$label',
+            options: (query) => {
+              return new Promise((resolve, reject) => {
```
Comment (bvkimball) [inline src/app/pages/elements/form/FormDemo.ts]: return Promise.resolve(this.getCountryOptions(query));

Comment (amrutha-rajiv) [reply src/app/pages/elements/form/FormDemo.ts]: done

**Comment by bvkimball** [comment]
Looking Good!

**Comment by jgodi** [comment]
|File (6)|Change (:white_check_mark: +0.04%)|
|---|---|
|FormValidators.ts|:bangbang: -1.75%|
|Address.ts|:white_check_mark: +0.06%|
|Picker.ts|:bangbang: -0.73%|
|novo-label-service.ts|:white_check_mark: +0.35%|
|Countries.ts|:bangbang: -3.57%|
|FormUtils.ts|:bangbang: -0.30%|

*Please note, individual file changes won't add up to the overall coverage due to number of lines!*

**All new files must be at 80% coverage and old files must NOT decrease in coverage!**

**Comment by jgodi** [comment]
|File (6)|Change (:white_check_mark: +0.04%)|
|---|---|
|FormValidators.ts|:bangbang: -1.75%|
|Address.ts|:white_check_mark: +0.06%|
|Picker.ts|:bangbang: -0.73%|
|novo-label-service.ts|:white_check_mark: +0.35%|
|Countries.ts|:bangbang: -3.57%|
|FormUtils.ts|:bangbang: -0.30%|

*Please note, individual file changes won't add up to the overall coverage due to number of lines!*

**All new files must be at 80% coverage and old files must NOT decrease in coverage!**

**Pull request merged**